src/tools/clippy/clippy_utils/src/hir_utils.rs

```diff
@@ -97,7 +97,7 @@ pub struct HirEqInterExpr<'a, 'b, 'tcx> {
 impl HirEqInterExpr<'_, '_, '_> {
     pub fn eq_stmt(&mut self, left: &Stmt<'_>, right: &Stmt<'_>) -> bool {
         match (&left.kind, &right.kind) {
-            (&StmtKind::Local(l), &StmtKind::Local(r)) => {
+            (&StmtKind::Local(l, _), &StmtKind::Local(r, _)) => {
```

Comment:
In the `has_stmt` function the second field is used for the hash. Here it isn't. This is a bug. Can you please also add an equality  check for the second field here?

Reply:
Got it. The missing equality checks are added.